lib/OpenQA/Worker/Engines/isotovideo.pm

```diff
@@ -466,7 +467,9 @@ sub _engine_workit_step_2 ($job, $job_settings, $vars, $shared_cache, $callback)
             local $ENV{PERL5OPT} = '';
             # Allow to override isotovideo executable with an arbitrary
             # command line based on a config option
-            exec $job_settings->{ISOTOVIDEO} ? $job_settings->{ISOTOVIDEO} : ('perl', $isotovideo, '-d');
+            my $args = ('-d');
+            push @{$args}, '--color=yes' if $isotovideo_interface_version > 26;
+            exec $job_settings->{ISOTOVIDEO} ? $job_settings->{ISOTOVIDEO} : ('perl', $isotovideo, $args);
```

Comment:
I'm wondering whether this will work but I suppose if `$args` was actually an array (`@args`) is would.

Reply:
See above. I should apparently not trust our unit tests too much...

Reply:
I guess the test passes because the `push @{$args}` is never executed, so basically it is a string `$args = '-d'` and this is expected to work.
`push @$args` on a string would die under `strict`.

Reply:
All worker* tests pass for me locally even with an unconditional push

Reply:
I suppose tests should go into `t/24-worker-engine.t` (even though it doesn't cover this particular code yet).